Requires netCDF 4.7.4 or greater, but 4.7.4 is already installed

Hello,

I am trying to build version 4.5.3 (as well as dev version) on Windows 10. I am linking against netCDF-C 4.7.4 which I initially installed on my machine, but also successfully built as a static library. The compilers I am using are MSVC 19.27.29110 and Intel Fortran 19.1.2.254. 

Running "configure" in cmake 3.18.0, I get an error about 
```netcdf-c version 4.7.4 or greater is required```
which I believe is triggered by not finding nc_def_var_szip in the lib file. 
I managed to go around by commenting the line triggering the fatal error.

Proceeding with building the static netcdff, all seems to go well. However, it does not seem to contain any nf90_* symbols. Subsequently, trying to build any of the tests, it complains about not finding these symbols.

This problem is still present in v4.6.1 (with netcdf-c v4.9.2). Possibly related to using libaec rather than szip?

```
-- Performing Test HAVE_SZIP_WRITE
-- Performing Test HAVE_SZIP_WRITE - Success
-- Performing Test HAVE_QUANTIZE
-- Performing Test HAVE_QUANTIZE - Success
-- Looking for nc_def_var_szip in netCDF::netcdf
-- Looking for nc_def_var_szip in netCDF::netcdf - not found
CMake Error at CMakeLists.txt:668 (message):
  netcdf-c version 4.7.4 or greater is required
```

>  Out of curiosity, @mjs271 do you have an older version of libnetcdf on your system that the cmake-based build was picking up?

It seems unlikely that this is the case. The machine does have a module system, but the only available `netcdf-c` is v4.8.1. Of course, it's always possible that there's a surprise version buried deep down somewhere, since it's an org-managed system and not my own.

Regarding the CMake retry, it's still failing with the same error, despite claiming to have found NetCDF-C. Here's my config script where I tried to provide all the information I could (hopefully not too much):
```shell
THIS_PATH=${HOME}/libs/netcdf-fortran/
export NetCDF_C_ROOT=${HOME}/libs/netcdf-c/install

export NETCDF_C_LIBRARY=${NetCDF_C_ROOT}/lib64
export NETCDF_C_INCLUDE_DIR=${NetCDF_C_ROOT}/include
export LD_LIBRARY_PATH="${NETCDF_C_LIBRARY}:${LD_LIBRARY_PATH}"
export PATH=${NetCDF_C_ROOT}/bin:${PATH}

export CPPFLAGS="-I${NETCDF_C_INCLUDE_DIR}"
export LDFLAGS="-L${NETCDF_C_LIBRARY}"

cmake ${THIS_PATH} \
  -DCMAKE_C_COMPILER=mpicc \
  -DCMAKE_CXX_COMPILER=mpicxx \
  -DCMAKE_Fortran_COMPILER=mpif90 \
  -DCMAKE_INSTALL_PREFIX=${THIS_PATH}/install\
  -DCMAKE_PREFIX_PATH=${THIS_PATH}/install \
  -DCMAKE_NETCDF_C_PATH=${NetCDF_C_ROOT} \
  -DCMAKE_NETCDF_C_LIBRARY=${NETCDF_C_LIBRARY} \
  -DCMAKE_NETCDF_C_INCLUDE_DIR=${NETCDF_C_LIBRARY} \
  -DBUILD_SHARED_LIBS=ON
```
For completeness, here's my build environment:
```shell
Currently Loaded Modulefiles:
  1) cmake/3.27.7
  2) gcc/10.3.0
  3) openmpi/4.1.2-gcc-10.3.0
  4) hdf5/1.10.6-gcc-10.3.0-openmpi-4.1.2
```

And here's the output relating to netCDF:
```shell
-- Found netCDF CMake package: netCDF::netcdf
-- Whether NetCDF-C built with HDF5 enabled: yes
[...]
-- Looking for nccreate in netCDF::netcdf
-- Looking for nccreate in netCDF::netcdf - not found
-- Looking for nc_set_log_level in netCDF::netcdf
-- Looking for nc_set_log_level in netCDF::netcdf - not found
-- Looking for oc_open in netCDF::netcdf
-- Looking for oc_open in netCDF::netcdf - not found
-- Performing Test HAVE_SZIP_WRITE
-- Performing Test HAVE_SZIP_WRITE - Failed
-- Performing Test HAVE_QUANTIZE
-- Performing Test HAVE_QUANTIZE - Failed
-- Looking for nc_def_var_szip in netCDF::netcdf
-- Looking for nc_def_var_szip in netCDF::netcdf - not found
CMake Error at CMakeLists.txt:672 (message):
  netcdf-c version 4.7.4 or greater is required
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -669,7 +669,7 @@ endif()
 
 CHECK_LIBRARY_EXISTS(${NETCDF_C_LIBRARY} nc_def_var_szip "" HAVE_DEF_VAR_SZIP)
 if (NOT HAVE_DEF_VAR_SZIP)
-  message(FATAL_ERROR "netcdf-c version 4.7.4 or greater is required")
+  message(FATAL_ERROR "netcdf-c version 4.7.4 or greater is required, built with underlying szip support.")
 endif()
 
 ###

--- a/configure.ac
+++ b/configure.ac
@@ -553,7 +553,7 @@ AC_CHECK_FUNCS([nc_def_opaque nccreate nc_set_log_level oc_open nc_def_var_szip]
 
 # Starting with version 4.7.4, netcdf-c has function nc_def_var_szip().
 if test "x$ac_cv_func_nc_def_var_szip" != xyes; then
-   AC_MSG_ERROR([netcdf-c version 4.7.4 or greater is required])
+   AC_MSG_ERROR([netcdf-c version 4.7.4 or greater is required, built with underlying szip support.])
 fi
 
 nc_build_v2=no

